rezz.lic

```diff
@@ -1,118 +1,131 @@
-    rejuv(person)
-
-    # gesture
-    bput("gest #{person}", 'Roundtime')
+  
+  def check_favor
+    Flags.add('no-favor', /(#{@person}.*) has no favor with (.*) god/)
+    bput("perc #{@person}", /#{@person} has (.*) favors with (his|her) god/)
+    Flags['no-favor'] ? begin echo "*** #{@person} has no favors! ***"; exit end : nil
   end
-
-  def find_soul(person, infuse_amt)
-    loop do
-      pause 5 while mana <= 40
-      harness_mana([infuse_amt]) unless skip_harness?
-      case bput("infuse rezz #{infuse_amt}", /You clearly sense the spirit of #{person} in the Void/i, 'You sense a spirit nearby, but you are unable to make it out clearly.', 'Your sense of sight has not fully transitioned to the spirit world', "You don't have enough harnessed mana to infuse that much.")
-      when 'You sense a spirit nearby, but you are unable to make it out clearly.'
-        next
-      when /You clearly sense the spirit of #{person} in the Void/i
-        break
-      when 'Your sense of sight has not fully transitioned to the spirit world'
-        pause 2
-        next
-      when "You don't have enough harnessed mana to infuse that much."
-        next
-      else
-        next
-      end
+  
+  def rejuv(quick = false)
+    quick ? begin @rejuv['cambrinth'] = []; @rejuv['prep_time'] = 5; @rejuv['mana'] = 5 end : nil
+    Flags.add('wrong-name', "You can't cast that at yourself")
+    Flags.add('silver', 'A thin silver nimbus surrounds')
+    Flags.add('alive', 'Nothing happens')
+    until Flags['silver'] || Flags['alive'] || Flags['wrong-name']
+	  pause 1 while DRStats.mana < 50
+      cast_spell(@rejuv, @settings)
+  	  pause 0.5
     end
+    Flags['alive'] ? begin echo "*** #{@person} is already alive, dumbass! ***"; exit end : nil
+    Flags['wrong-name'] ? begin echo "*** Name spelled incorrectly. ***"; exit end : nil
+    Flags.reset('rejuv-silver')
   end
-
-  def is_silver?(person)
-    /As you concentrate, a thin (.*) nimbus/ =~ bput("perc #{person}", 'As you concentrate, a thin (.*) nimbus')
-    color = Regexp.last_match(1)
-    echo "Memories are #{color}"
-    waitrt?
-    color.eql? 'silver'
-  end
-
-  def rejuv(person)
-    return if is_silver?(person)
-    data = get_data('spells').spell_data['Rejuvenation']
-    data['cast'] = "cast #{person}"
-    if @rejuv_mana.zero?
-      data = check_discern(data, @settings)
-      @rejuv_mana = data['mana']
+    
+  def cast_rezz
+    Flags.add('rezz-ready', 'You prepare to look for the lost spirits of the dead')
+    Flags.add('rezz-not-prepped', 'This pattern may only be cast with full preparation')
+    cast_spell(@rezz, @settings)
+    if Flags['rezz-not-prepped']
+      while Flags['rezz-not-prepped'] 
+	    Flags.reset('rezz-not-prepped')
+        pause 1
+	    bput('cast', get_data('spells').cast_messages, )
+	  end
     end
-
-    loop do
-      cast_spell(data, @settings)
-      break if is_silver?(person)
+    while !Flags['rezz-ready']
```

Comment:
You can use `until` instead of `while not`. Rubocop may auto correct this

Reply:
I don't have rubocop installed, working on it.

Reply:
Don't worry about it, I was hesitant to point out style things because rubocop is going to correct it all in the end. I thought you might be interested though :)

Reply:
I notice there are a TON of review comments in here, so ill refrain to avoid fatiguing you :P But do check out the Ruby Style Guide